### DISCUSSION TITLE
ISR for route handlers in the /app directory

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#46649] [Help]
### Summary

Hello, I wrote a route handler in the beta `/app` directory and am wondering if we can revalidate the route manually instead of setting a revalidation time.

The route handler dynamically generates a sitemap by calling an API endpoint that prints all the page paths we used to generate the site. The data fetching uses the `fetch` function internally.

I tried using an endpoint in `/pages/api/` that runs `res.revalidate("/sitemap.xml")` but got a 500 status with no error message.

Is this supported or on the roadmap?

**Links:** [Route handler](https://github.com/mintlify/mint/blob/main/client/src/app/_sites/%5Bsubdomain%5D/(seo)/sitemap.xml/route.ts), [revalidation endpoint](https://github.com/mintlify/mint/blob/main/client/src/pages/api/revalidate.ts)

**Comment by wwwdepot** [comment]
`res.revalidate` bypass middleware. ([Read the note section](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation)) 

So you will need to call the actual path `res.revalidate("/actual/path/to/sitemap.xml")`

```js
// bad
res.revalidate("/sitemap.xml")

// good
res.revalidate("/_sites/actual.domain.com/sitemap.xml")
```

**Reply by ghost** [reply]
Thank you for checking out the repo! I am passing the full `/_sites/actual.domain.com/sitemap.xml` path and get the 500 error when doing so. I shortened it for the PR description because I hadn't explained that we had a middleware.

Revalidating other pages with the `/_sites/actual.domain.com/` at the start works well, so I'm guessing the issue has to be something about route handlers rather than the middleware code.
